Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp

```diff
@@ -103,8 +99,14 @@ void WebPermissionController::tryProcessingRequests()
             continue;
         }
 
+        auto* webPage = WebProcess::singleton().webPage(currentRequest.identifier);
```

Comment:
I think Sihui recommended this but I disagree with the suggestion:
1. The request should store the WebPageProxyIdentifier, not the PageIdentifier
2. There is no point in looking up the WebPage here if you already have the right identifier you need to send over IPC
3. This adds more complexity to try and avoid an IPC round trip in a very edge case where the WebPage doesn't exist. This is just not worth it.
4. Added more WebPage-related code is not a good idea because we're going to support service workers, which won't have a WebPage.

Reply:
I don't think this is an edge case. In current impl, we block request handling when there's a pending request (waiting for UI process's reply). If a page makes multiple query requests and is gone, it's going to block requests from other pages for multiple rounds of IPCs.
I actually think we should store std::optional<WeakPtr<WebPage>> in the request, but PageIdentifier or WeakPtr<WebPage> might be enough for this patch.

Reply:
Very edge case. Multiple requests in parallel is an edge case, multiple pages in a single WebProcess is already an edge case. Even if there were really several and some queueing were to happen, querying doesn't involve prompting and we can thus expect responses really fast.

Also, an IPC round trip is cheap.

Reply:
I think you are adding complexity to try and optimize out a quick IPC round trip (quick since the UIProcess won't find the page and reply right away) for the case where:
1. There is more than one page in the WebProcess
and
2. Several of these WebPages are making query permission request in parallel
and
3. One of the WebPage gets closed before getting a response to its request(s)

Reply:
We've gone back to using WebPageProxyIdentifier here. Fixed!